src/tensors/sectorvector.jl

```diff
@@ -16,6 +16,11 @@ function SectorVector{T}(::UndefInitializer, V::ElementarySpace) where {T}
     structure = diagonalblockstructure(V ← V)
     return SectorVector(data, structure)
 end
+function SectorVector{T, I, A}(::UndefInitializer, V::ElementarySpace) where {T, I, A <: AbstractVector{T}}
+    data = A(undef, reduceddim(V))
```

Comment:
I know we discussed whether `A <: DenseVector` was desirable in the `SectorVector` type definition. For this constructor, I think that if not `A<:DenseVector`, it is unlikely that `A(undef, reduceddim(V))` is going to be a valid constructor call, but I am fine leaving this as is.